feature/add_location_table_to_country_report (#42)
fivetran/dbt_tiktok_ads: CHANGELOG.md

# dbt_tiktok_ads v1.3.0

[PR #42](https://github.com/fivetran/dbt_tiktok_ads/pull/42) includes the following updates:

## Schema/Data Change

> **3 total changes • 0 possible breaking changes**

Reviewer: Could you update the number of materialized models in the README [here](https://github.com/fivetran/dbt_tiktok_ads/blob/main/README.md#resources)?

| Data Model | Change Type | Old | New | Notes |
|------------|-------------|-----|-----|-------|
| [stg_tiktok_ads__location_tmp](https://fivetran.github.io/dbt_tiktok_ads/#!/model/model.tiktok_ads.stg_tiktok_ads__location_tmp) | New Temp Model | | | Temp model added for the `location` source table. |

> | [stg_tiktok_ads__location](https://fivetran.github.io/dbt_tiktok_ads/#!/model/model.tiktok_ads.stg_tiktok_ads__location) | New Staging Model | | | Surfaces the reference table of geographic regions and locations available for TikTok Ads targeting. |

Reviewer: Honestly since we're making the changelogs more impact-focused, we could perhaps remove these lines and just mention in the notes for `tiktok_ads__campaign_country_report` that this adds 2 new materialized models if data is present

Author: Are we not doing schema change tables at all anymore? I still think these are useful for dbt Core users, but maybe without notes?

> | [tiktok_ads__campaign_country_report](https://fivetran.github.io/dbt_tiktok_ads/#!/model/model.tiktok_ads.tiktok_ads__campaign_country_report) | New Columns | | `region_name`, `area_type`, `parent_id`, `support_below_18` | Joins the new `stg_tiktok_ads__location` model on `country_code` to add geographic region attributes. |

Reviewer: I don't think we need to include this note

## Bug Fix

> - Replaces the invalid `"datetime"` datatype on `stat_time_day` with `dbt.type_timestamp()` in the `get_campaign_country_report_columns` macro. Databricks users hit a compilation error when the column was missing from the source table and `fill_staging_columns` generated a cast to the unsupported `datetime` type.

Reviewer: This is a bit technical..could you adjust to focus on it ensuring proper data types/avoiding failures for Databricks users missing the `campaign_country_report.stat_time_day` field?

Something like 

```suggestion
- Avoids potential data type failures for Databricks users not syncing the `campaign_country_report.stat_time_day` field.
```

# dbt_tiktok_ads v1.3.0

[PR #42](https://github.com/fivetran/dbt_tiktok_ads/pull/42) includes the following updates:

## Schema/Data Change
**3 total changes • 0 possible breaking changes**
| Data Model | Change Type | Old | New | Notes |
|------------|-------------|-----|-----|-------|
| [stg_tiktok_ads__location_tmp](https://fivetran.github.io/dbt_tiktok_ads/#!/model/model.tiktok_ads.stg_tiktok_ads__location_tmp) | New Tmp Model | | | |
| [stg_tiktok_ads__location](https://fivetran.github.io/dbt_tiktok_ads/#!/model/model.tiktok_ads.stg_tiktok_ads__location) | New Staging Model | | | |

> | [tiktok_ads__campaign_country_report](https://fivetran.github.io/dbt_tiktok_ads/#!/model/model.tiktok_ads.tiktok_ads__campaign_country_report) | New Columns | | `country_name`, `parent_location_id`, `has_support_below_18` | |

Reviewer: Since we're making the changelogs Quickstart-focused, I'd recommend only highlighting `tiktok_ads__campaign_country_report` here. We can mention that this change is supported by the newly included `LOCATION` table/2 staging models

Can also discuss this with the release reviewer if you disagree -- at the very least we should reverse the order of the rows so that tiktok_ads__campaign_country_report is first and stg_tiktok_ads__location_tmp is last

Author: Moved to feature update section and nested under "for dbt core users" section. Will ask for feedback from release reviewer.

## Feature Update

> - **For dbt Core users**: Adds the `tiktok_ads__using_location` variable (`true` by default). If you do not want to sync the `location` table from your TikTok Ads connection, set this variable to `false` in your `dbt_project.yml` to disable the location join and the related columns in `tiktok_ads__campaign_country_report`. See the [README](https://github.com/fivetran/dbt_tiktok_ads/blob/main/README.md#disable-location-enrichment) for more details.

Reviewer: ```suggestion
- **For dbt Core users**: Adds the `tiktok_ads__using_location` variable (`true` by default). If you do not sync or want to use the `location` table, set this variable to `false` in your `dbt_project.yml`. See the [README](https://github.com/fivetran/dbt_tiktok_ads/blob/main/README.md#disable-location-enrichment) for more details.
```

## Bug Fix

> - Avoids potential data type failures for Databricks users not syncing the `campaign_country_report.stat_time_day` field.  

Reviewer 2: ```suggestion
- Fixes potential data type failures for Databricks users not syncing the `campaign_country_report.stat_time_day` field.  
```
